clang/include/clang/Basic/Builtins.td

```diff
@@ -4707,6 +4707,12 @@ def HLSLIsinf : LangBuiltin<"HLSL_LANG"> {
   let Prototype = "void(...)";
 }
 
+def HLSLLength : LangBuiltin<"HLSL_LANG"> {
+  let Spellings = ["__builtin_hlsl_elementwise_length"];
```

Comment:
This shouldn't be an elementwise builtin see `lerp`, 'dot`, and `mad`. For operations that work on vectors and generate scalar results we don't add the elementwise naming convention. Reserve elementwise for  cases were the function takes multiple inputs applys the same operation to each input then returns the same number of outputs.